front/lib/api/assistant/conversation.ts

```diff
@@ -1502,6 +1504,7 @@ export async function* retryAgentMessage(
         actions: [],
         content: null,
         chainOfThought: null,
+        visualizations: [],
```

Comment:
I'm wondering whether we should create a more generic solution that can support capabilities beyond just visualization. I don't have specific ideas for other capabilities that might not need a dedicated action implementation right now, but I'm confident we will face this need in the future. Maybe we could have a dedicated table linking an `AgentMessageId` with the capability type and its result. WDYT?

Reply:
I understand the point. For now I feel a bit antsy about creating an abstraction I'm not yet sure we'll really need

Reply:
I also don't fully understand the separate table thing ? this is all stored in `AgentMessageContent`

Reply:
(which is already a separate table -- we don't store the extracted viz directly)

Reply:
Ok I missed the `AgentMessageContent`, which is good then wanted to make sure it does not live in the `agent_messages` table.

Reply:
Yeah it's just a field on the agent message type (as in output / API type) for convenience (so consumers of the API don't need to do the content parsing themselves)

Reply:
It's just that boolean values in SQL tables don't scale well, and we don't want to have many columns with this type. Hence, the proposal to simply use an array of strings instead.

Reply:
Yeah for the `visualizationEnabled` part it's another topic, array of strings could be better. I just wonder if we'll ever need to have other strings in that array. IMO it might be better to keep it as a boolean for now and only migrate to array of strings once we have the use-case for a second boolean